app/models/account.rb

```diff
@@ -155,6 +155,20 @@ def first_valuation_amount
     first_valuation&.amount_money || balance_money
   end
 
+  # Get short version of the subtype label
+  def short_subtype_label
+    return nil unless subtype && accountable_type
+    accountable_class = accountable_type.constantize
+    accountable_class.short_subtype_label_for(subtype) || accountable_class.display_name
+  end
+
+  # Get long version of the subtype label
+  def long_subtype_label
+    return nil unless subtype && accountable_type
+    accountable_class = accountable_type.constantize
+    accountable_class.long_subtype_label_for(subtype) || accountable_class.display_name
+  end
```

Comment:
Small nit, but `accountable_class` should automatically be available on each Account instance, so these could be simplified to:

```rb
def short_subtype_label
  accountable_class.short_subtype_label_for(subtype) || accountable_class.display_name
end

def long_subtype_label
  accountable_class.long_subtype_label_for(subtype) || accountable_class.display_name
end
```